File: src/game/phonicsData.ts
// Phonics game data - words organized by beginning sounds and difficulty
// Used for the phonics sound matching game

export interface PhonicsWord {
  word: string
  image: string // emoji placeholder - can be replaced with actual image URLs
  beginningSound: string // the letter sound to match (e.g., 'c' for 'cat')
  audio?: string // audio file path (optional)
}

export interface DifficultyLevel {
  name: string
  description: string
  words: PhonicsWord[]
  choiceCount: number // how many letter choices to show (3-4)
}

// Words grouped by beginning sound, then organized into difficulty levels
const wordsBySound: Record<string, PhonicsWord[]> = {
  c: [
    { word: 'cat', image: '🐱', beginningSound: 'c' },
    { word: 'car', image: '🚗', beginningSound: 'c' },
    { word: 'cup', image: '🥤', beginningSound: 'c' },
    { word: 'cow', image: '🐄', beginningSound: 'c' },
  ],
  d: [
    { word: 'dog', image: '🐕', beginningSound: 'd' },
    { word: 'duck', image: '🦆', beginningSound: 'd' },
    { word: 'door', image: '🚪', beginningSound: 'd' },
    { word: 'drum', image: '🥁', beginningSound: 'd' },
  ],
  f: [
    { word: 'fish', image: '🐟', beginningSound: 'f' },
    { word: 'fox', image: '🦊', beginningSound: 'f' },
    { word: 'frog', image: '🐸', beginningSound: 'f' },
    { word: 'fan', image: '🪭', beginningSound: 'f' },
  ],
  s: [
    { word: 'sun', image: '☀️', beginningSound: 's' },
    { word: 'star', image: '⭐', beginningSound: 's' },
    { word: 'snake', image: '🐍', beginningSound: 's' },
    { word: 'sock', image: '🧦', beginningSound: 's' },
  ],
  b: [
    { word: 'ball', image: '⚽', beginningSound: 'b' },
    { word: 'bear', image: '🐻', beginningSound: 'b' },
    { word: 'bus', image: '🚌', beginningSound: 'b' },
    { word: 'bird', image: '🐦', beginningSound: 'b' },
  ],
  m: [
    { word: 'moon', image: '🌙', beginningSound: 'm' },
    { word: 'mouse', image: '🐭', beginningSound: 'm' },
    { word: 'map', image: '🗺️', beginningSound: 'm' },
    { word: 'milk', image: '🥛', beginningSound: 'm' },
  ],
  p: [
    { word: 'pig', image: '🐷', beginningSound: 'p' },
    { word: 'pizza', image: '🍕', beginningSound: 'p' },
    { word: 'pen', image: '🖊️', beginningSound: 'p' },
    { word: 'pear', image: '🍐', beginningSound: 'p' },
  ],
  t: [
    { word: 'tree', image: '🌳', beginningSound: 't' },
    { word: 'turtle', image: '🐢', beginningSound: 't' },
    { word: 'train', image: '🚂', beginningSound: 't' },
    { word: 'tent', image: '⛺', beginningSound: 't' },
  ],
  h: [
    { word: 'hat', image: '🎩', beginningSound: 'h' },
    { word: 'house', image: '🏠', beginningSound: 'h' },
    { word: 'heart', image: '❤️', beginningSound: 'h' },
    { word: 'horse', image: '🐴', beginningSound: 'h' },
  ],
  r: [
    { word: 'rain', image: '🌧️', beginningSound: 'r' },
    { word: 'ring', image: '💍', beginningSound: 'r' },
    { word: 'rabbit', image: '🐰', beginningSound: 'r' },
    { word: 'rose', image: '🌹', beginningSound: 'r' },
  ],
}

// Get all available letter sounds
export function getAvailableSounds(): string[] {
  return Object.keys(wordsBySound)
}

// Get all words for a specific sound
export function getWordsBySound(sound: string): PhonicsWord[] {
  return wordsBySound[sound.toLowerCase()] || []
}

// Difficulty levels with progressive challenge
export const difficultyLevels: DifficultyLevel[] = [
  {
    name: 'Easy',
    description: 'Common sounds with 3 choices',
    choiceCount: 3,
    words: [
      ...wordsBySound.c.slice(0, 2),
      ...wordsBySound.d.slice(0, 2),
      ...wordsBySound.s.slice(0, 2),
      ...wordsBySound.b.slice(0, 2),
    ],
  },
  {
    name: 'Medium',
    description: 'More sounds with 3 choices',
    choiceCount: 3,
    words: [
      ...wordsBySound.f.slice(0, 2),
      ...wordsBySound.m.slice(0, 2),
      ...wordsBySound.p.slice(0, 2),
      ...wordsBySound.t.slice(0, 2),
    ],
  },
  {
    name: 'Hard',
    description: 'All sounds with 4 choices',
    choiceCount: 4,
    words: [
      ...wordsBySound.c.slice(2),
      ...wordsBySound.d.slice(2),
      ...wordsBySound.f.slice(2),
      ...wordsBySound.s.slice(2),
      ...wordsBySound.h,
      ...wordsBySound.r,
    ],
  },
]

// Get a random word from a specific level
export function getRandomWordFromLevel(levelIndex: number): PhonicsWord {
  const level = difficultyLevels[levelIndex] || difficultyLevels[0]
  return level.words[Math.floor(Math.random() * level.words.length)]
}

// Get wrong answer choices (letters that are NOT the correct answer)
export function getWrongChoices(
  correctSound: string,
  count: number
): string[] {
  const allSounds = getAvailableSounds()
  const wrongSounds = allSounds.filter((s) => s !== correctSound.toLowerCase())

  // Shuffle and take the needed count
  const shuffled = [...wrongSounds].sort(() => Math.random() - 0.5)
  return shuffled.slice(0, count)
}

// Get all choices (correct + wrong, shuffled)
export function getShuffledChoices(
  correctSound: string,
  totalChoices: number
): string[] {
  const wrongChoices = getWrongChoices(correctSound, totalChoices - 1)
  const allChoices = [correctSound.toLowerCase(), ...wrongChoices]
  return allChoices.sort(() => Math.random() - 0.5)
}

// Get total word count across all levels
export function getTotalWordCount(): number {
  return difficultyLevels.reduce((sum, level) => sum + level.words.length, 0)
}

// Get word count for a specific level
export function getLevelWordCount(levelIndex: number): number {
  const level = difficultyLevels[levelIndex]
  return level ? level.words.length : 0
}

File: ios/L2RR2L/Features/PhonicsGame/Models/PhonicsData.swift
import Foundation

/// Static phonics word data for the game
enum PhonicsData {
    /// All available beginning sounds
    static let beginningSounds = ["b", "c", "d", "f", "g", "h", "m", "n", "p", "r", "s", "t"]

    /// All phonics words
    static let words: [PhonicsWord] = [
        PhonicsWord(id: "ball", word: "ball", beginningSound: "b", phonemes: ["b", "all"], difficulty: 1, image: "/assets/phonics/ball.svg", emoji: "soccer ball", audio: "/assets/audio/ball.mp3", category: "toys"),
        PhonicsWord(id: "bed", word: "bed", beginningSound: "b", phonemes: ["b", "e", "d"], difficulty: 1, image: "/assets/phonics/bed.svg", emoji: "bed", audio: "/assets/audio/bed.mp3", category: "home"),
        PhonicsWord(id: "bus", word: "bus", beginningSound: "b", phonemes: ["b", "u", "s"], difficulty: 1, image: "/assets/phonics/bus.svg", emoji: "bus", audio: "/assets/audio/bus.mp3", category: "vehicles"),
        PhonicsWord(id: "cat", word: "cat", beginningSound: "c", phonemes: ["c", "a", "t"], difficulty: 1, image: "/assets/phonics/cat.svg", emoji: "cat", audio: "/assets/audio/cat.mp3", category: "animals"),
        PhonicsWord(id: "cup", word: "cup", beginningSound: "c", phonemes: ["c", "u", "p"], difficulty: 1, image: "/assets/phonics/cup.svg", emoji: "cup with straw", audio: "/assets/audio/cup.mp3", category: "home"),
        PhonicsWord(id: "car", word: "car", beginningSound: "c", phonemes: ["c", "ar"], difficulty: 2, image: "/assets/phonics/car.svg", emoji: "automobile", audio: "/assets/audio/car.mp3", category: "vehicles"),
        PhonicsWord(id: "dog", word: "dog", beginningSound: "d", phonemes: ["d", "o", "g"], difficulty: 1, image: "/assets/phonics/dog.svg", emoji: "dog", audio: "/assets/audio/dog.mp3", category: "animals"),
        PhonicsWord(id: "duck", word: "duck", beginningSound: "d", phonemes: ["d", "u", "ck"], difficulty: 2, image: "/assets/phonics/duck.svg", emoji: "duck", audio: "/assets/audio/duck.mp3", category: "animals"),
        PhonicsWord(id: "fish", word: "fish", beginningSound: "f", phonemes: ["f", "i", "sh"], difficulty: 1, image: "/assets/phonics/fish.svg", emoji: "fish", audio: "/assets/audio/fish.mp3", category: "animals"),
        PhonicsWord(id: "fan", word: "fan", beginningSound: "f", phonemes: ["f", "a", "n"], difficulty: 2, image: "/assets/phonics/fan.svg", emoji: "cyclone", audio: "/assets/audio/fan.mp3", category: "home"),
        PhonicsWord(id: "fox", word: "fox", beginningSound: "f", phonemes: ["f", "o", "x"], difficulty: 2, image: "/assets/phonics/fox.svg", emoji: "fox", audio: "/assets/audio/fox.mp3", category: "animals"),
        PhonicsWord(id: "hat", word: "hat", beginningSound: "h", phonemes: ["h", "a", "t"], difficulty: 1, image: "/assets/phonics/hat.svg", emoji: "top hat", audio: "/assets/audio/hat.mp3", category: "clothing"),
        PhonicsWord(id: "hen", word: "hen", beginningSound: "h", phonemes: ["h", "e", "n"], difficulty: 2, image: "/assets/phonics/hen.svg", emoji: "chicken", audio: "/assets/audio/hen.mp3", category: "animals"),
        PhonicsWord(id: "house", word: "house", beginningSound: "h", phonemes: ["h", "ou", "se"], difficulty: 3, image: "/assets/phonics/house.svg", emoji: "house", audio: "/assets/audio/house.mp3", category: "home"),
        PhonicsWord(id: "map", word: "map", beginningSound: "m", phonemes: ["m", "a", "p"], difficulty: 1, image: "/assets/phonics/map.svg", emoji: "world map", audio: "/assets/audio/map.mp3", category: "objects"),
        PhonicsWord(id: "moon", word: "moon", beginningSound: "m", phonemes: ["m", "oo", "n"], difficulty: 2, image: "/assets/phonics/moon.svg", emoji: "crescent moon", audio: "/assets/audio/moon.mp3", category: "nature"),
        PhonicsWord(id: "net", word: "net", beginningSound: "n", phonemes: ["n", "e", "t"], difficulty: 2, image: "/assets/phonics/net.svg", emoji: "goal net", audio: "/assets/audio/net.mp3", category: "objects"),
        PhonicsWord(id: "nut", word: "nut", beginningSound: "n", phonemes: ["n", "u", "t"], difficulty: 2, image: "/assets/phonics/nut.svg", emoji: "peanuts", audio: "/assets/audio/nut.mp3", category: "food"),
        PhonicsWord(id: "pig", word: "pig", beginningSound: "p", phonemes: ["p", "i", "g"], difficulty: 1, image: "/assets/phonics/pig.svg", emoji: "pig face", audio: "/assets/audio/pig.mp3", category: "animals"),
        PhonicsWord(id: "pot", word: "pot", beginningSound: "p", phonemes: ["p", "o", "t"], difficulty: 2, image: "/assets/phonics/pot.svg", emoji: "pot of food", audio: "/assets/audio/pot.mp3", category: "home"),
        PhonicsWord(id: "sun", word: "sun", beginningSound: "s", phonemes: ["s", "u", "n"], difficulty: 1, image: "/assets/phonics/sun.svg", emoji: "sun", audio: "/assets/audio/sun.mp3", category: "nature"),
        PhonicsWord(id: "sock", word: "sock", beginningSound: "s", phonemes: ["s", "o", "ck"], difficulty: 2, image: "/assets/phonics/sock.svg", emoji: "socks", audio: "/assets/audio/sock.mp3", category: "clothing"),
        PhonicsWord(id: "star", word: "star", beginningSound: "s", phonemes: ["s", "t", "ar"], difficulty: 3, image: "/assets/phonics/star.svg", emoji: "star", audio: "/assets/audio/star.mp3", category: "nature"),
        PhonicsWord(id: "top", word: "top", beginningSound: "t", phonemes: ["t", "o", "p"], difficulty: 1, image: "/assets/phonics/top.svg", emoji: "TOP arrow", audio: "/assets/audio/top.mp3", category: "objects"),
        PhonicsWord(id: "ten", word: "ten", beginningSound: "t", phonemes: ["t", "e", "n"], difficulty: 2, image: "/assets/phonics/ten.svg", emoji: "keycap: 10", audio: "/assets/audio/ten.mp3", category: "numbers"),
        PhonicsWord(id: "tree", word: "tree", beginningSound: "t", phonemes: ["t", "r", "ee"], difficulty: 3, image: "/assets/phonics/tree.svg", emoji: "deciduous tree", audio: "/assets/audio/tree.mp3", category: "nature"),
        PhonicsWord(id: "rug", word: "rug", beginningSound: "r", phonemes: ["r", "u", "g"], difficulty: 2, image: "/assets/phonics/rug.svg", emoji: "brown square", audio: "/assets/audio/rug.mp3", category: "home"),
        PhonicsWord(id: "rain", word: "rain", beginningSound: "r", phonemes: ["r", "ai", "n"], difficulty: 3, image: "/assets/phonics/rain.svg", emoji: "cloud with rain", audio: "/assets/audio/rain.mp3", category: "nature"),
        PhonicsWord(id: "goat", word: "goat", beginningSound: "g", phonemes: ["g", "oa", "t"], difficulty: 2, image: "/assets/phonics/goat.svg", emoji: "goat", audio: "/assets/audio/goat.mp3", category: "animals"),
        PhonicsWord(id: "gift", word: "gift", beginningSound: "g", phonemes: ["g", "i", "f", "t"], difficulty: 3, image: "/assets/phonics/gift.svg", emoji: "wrapped gift", audio: "/assets/audio/gift.mp3", category: "objects")
    ]

    /// Get words filtered by difficulty
    static func words(forDifficulty difficulty: Int) -> [PhonicsWord] {
        words.filter { $0.difficulty <= difficulty }
    }

    /// Get words that start with a specific sound
    static func words(startingWith sound: String) -> [PhonicsWord] {
        words.filter { $0.beginningSound == sound }
    }

    /// Get random words from the pool
    static func randomWords(count: Int, maxDifficulty: Int = 3) -> [PhonicsWord] {
        let pool = words(forDifficulty: maxDifficulty)
        return Array(pool.shuffled().prefix(count))
    }
}
